# Bioinformatics

here you can find some code which is for my bioinformatic project.  

### P2: (q2.1 file is for local-global allignment) 

  P2.2: To extract the genes of any of Ebola virus, we intend to use the extracted genes of the closest known species known to Ebola virus, Marburg. To do this, it is necessary that each of the Marburg genes is separately aligned with the entire genome of each Ebola species, so that a part of the genome most closely related to each of the genes is assumed to be the same gene in that species. Given that the length of a gene is not comparable with the length of the genome, the global alignment algorithm needs to be modified so that the cost of the initial and the end gaps are only considered 0 for the gene sequence (in fact, the algorithm is between the global and local alignment )
  
  P2.3: global allignment & Edit distance usinf the Needleman-Wunsch algorithm 
  
### P3: It is about creating UPGMA and NJ

  P3.1: Phylogenetic tree
  
  P3.2: Combine the trees by using the Consensus method
  
  P3.3: Compare the tree derived from Section 3.2 with a tree that is derived from the edit distance of species in global alignment.
  
  P3.4: To know what the first Ebola virus was, we need to get help from another creature that is evolving close to this ebulla virus, our       candidate for this is Marburg Hemorrhagic, then create UPGMA and NJ for these 6 species.
  
### P4:

  P4.1: Find the time interval between one species with another species & Estimate when the species is separated from their ancestors
  
  P4.2: Finding a probability distribution from the time that the next creation will be created
